app/javascript/src/controller_pages.js

```diff
@@ -215,23 +216,28 @@ function addQuestionMenuListeners(view) {
 
   // QuestionMenuSelectionRemove
   view.$document.on("QuestionMenuSelectionRemove", function(event, question) {
-    var html = $(templateContent).filter("[data-node=remove]").text();
-    view.dialogConfirmationDelete.open({
-      heading: html.replace(/#{label}/, question.$heading.text()),
-      ok: view.text.dialogs.button_delete_component
-      }, function() {
+    var questionUuid = question.data._uuid;
+    var apiUrl = question.menu.selectedItem.data('apiPath').replace('question_uuid', questionUuid);
```

Comment:
Assuming you have a path `/blah/blah/question_uuid/blah` (or something to that effect), where you're String.replacing the 'question_uuid' bit with a value.

I would suggest, if we can, we make the apiPath value more obvious by constructing its initial value in form `/blah/blah/#{question_uuid}/blah` and then search/replacing in form `str.replace(/#{question_uuid}/, question_uuid)`.

Obviously it doesn't make any difference to outcome but, looking at the lines being taken out, it would suggest we're already doing value replacements with that form of lookup identifier already (albeit with HTML in that example), so good idea to follow with same approach for consistency (sure there are others in and around code as well although, without documentation or any rules-based coding, appreciate it would not be easy to know this).

Note: I believe `#{something}` form is based on Ruby/Rails syntax although also commonly found in other languages and environments so hopefully a familiar thing many.